packages/curve-ui-kit/src/shared/ui/PositionDetails/HealthBar.tsx

```diff
@@ -0,0 +1,102 @@
+import { Box, Typography, type Theme } from '@mui/material'
+import { t } from '@ui-kit/lib/i18n'
+import { Reds, Blues } from '@ui-kit/themes/design/0_primitives'
+
+type HealthBarProps = {
+  health: number | undefined | null
+}
+
+const BAR_HEIGHT = '1.4375rem' // 23px
+const LINE_WIDTH = '0.25rem' // 4px
+const LABEL_GAP = '0.125rem' // 2px
+
+type LineColor = 'red' | 'orange' | 'green' | 'dark-green'
+
+const getLineColor = (color: LineColor) => (t: Theme) => {
+  switch (color) {
+    case 'red':
+      return t.design.Color.Tertiary[600]
+    case 'orange':
+      return t.design.Color.Tertiary[400]
+    case 'dark-green':
+      return t.design.Color.Secondary[600]
+    case 'green':
+      return t.design.Color.Secondary[500]
+  }
+}
+
+const Line = ({ first, position, color }: { first?: boolean; position: string; color: LineColor }) => (
+  <Box
+    sx={{
+      position: 'absolute',
+      top: 0,
+      left: first ? '0' : `calc(${position} - (${LINE_WIDTH} / 2))`,
+      width: LINE_WIDTH,
+      height: '100%',
+      backgroundColor: getLineColor(color),
+    }}
+  />
+)
+
+const Label = ({
+  first,
+  last,
+  position,
+  text,
+}: {
+  first?: boolean
+  last?: boolean
+  position: string
+  text: string
+}) => (
+  <Box
+    sx={{
+      position: 'absolute',
+      left: first || last ? position : `calc(${position} - (${LINE_WIDTH} / 2))`,
+      top: 0,
+      transform: last ? 'translateX(-100%)' : 'none',
+    }}
+  >
+    <Typography variant="bodyXsRegular" color="textTertiary" sx={{ whiteSpace: 'nowrap' }}>
+      {text}
+    </Typography>
+  </Box>
+)
+
+export const HealthBar = ({ health }: HealthBarProps) => {
+  const healthPercentage = Math.max(0, Math.min(health ?? 0, 100))
```

Comment:
This deserves a comment, plus: should it really happen in the component?
IMO it's store/query/hook logic. It can end up looking inconsistent across components.